packages/amplify-category-notifications/lib/multi-env-manager.js

```diff
@@ -183,87 +184,92 @@ async function pushChanges(context, pinpointNotificationsMeta) {
-                                serviceMeta.output.Id) {
-      availableChannels.forEach((channel) => {
-        if (serviceMeta.output[channel] && serviceMeta.output[channel].Enabled) {
-          enabledChannels.push(channel);
-        }
-      });
-      pinpointMeta = {
-        serviceName: services[i],
-        service: serviceMeta.service,
-        channels: enabledChannels,
-        Name: serviceMeta.output.Name,
-        Id: serviceMeta.output.Id,
-        Region: serviceMeta.output.Region,
-      };
-      break;
+  if (categoryMeta) {
+    const availableChannels = notificationManager.getAvailableChannels();
+    const enabledChannels = [];
+    const services = Object.keys(categoryMeta);
+    for (let i = 0; i < services.length; i++) {
+      const serviceMeta = categoryMeta[services[i]];
+      if (serviceMeta.service === constants.PinpointName &&
+                                  serviceMeta.output &&
+                                  serviceMeta.output.Id) {
+        availableChannels.forEach((channel) => {
+          if (serviceMeta.output[channel] && serviceMeta.output[channel].Enabled) {
+            enabledChannels.push(channel);
+          }
+        });
+        pinpointMeta = {
+          serviceName: services[i],
+          service: serviceMeta.service,
+          channels: enabledChannels,
+          Name: serviceMeta.output.Name,
+          Id: serviceMeta.output.Id,
+          Region: serviceMeta.output.Region,
+        };
+        break;
+      }
     }
   }
+  writeTeamProviderInfo(pinpointMeta, context);
+  writeBackendConfig(pinpointMeta, context.amplify.pathManager.getBackendConfigFilePath());
+  writeBackendConfig(pinpointMeta, context.amplify.pathManager.getCurrentBackendConfigFilePath());
+  writeAmplifyMeta(categoryMeta, context.amplify.pathManager.getAmplifyMetaFilePath());
+  writeAmplifyMeta(categoryMeta, context.amplify.pathManager.getCurentAmplifyMetaFilePath());
+  context.amplify.onCategoryOutputsChange(context);
+}
 
-  if (pinpointMeta) {
-    const teamProviderInfoFilepath = context.amplify.pathManager.getProviderInfoFilePath();
-    if (fs.existsSync(teamProviderInfoFilepath)) {
-      const teamProviderInfo = JSON.parse(fs.readFileSync(teamProviderInfoFilepath));
-      teamProviderInfo[envName] = teamProviderInfo[envName] || {};
-      teamProviderInfo[envName].categories = teamProviderInfo[envName].categories || {};
-      teamProviderInfo[envName].categories[constants.CategoryName] =
-                teamProviderInfo[envName].categories[constants.CategoryName] || {};
-      teamProviderInfo[envName].categories[constants.CategoryName][constants.PinpointName] = {
-        Name: pinpointMeta.Name,
-        Id: pinpointMeta.Id,
-        Region: pinpointMeta.Region,
-      };
-      const jsonString = JSON.stringify(teamProviderInfo, null, 4);
-      fs.writeFileSync(teamProviderInfoFilepath, jsonString, 'utf8');
-    }
+function writeTeamProviderInfo(pinpointMeta, context) {
```

Comment:
This should be moved to `amplify-cli`. This package should not know/care about what is the structure and location of teamProvider.

Reply:
agreed, when the cli exposes a method that write to a specific environment in the team provider info.